metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -380,36 +441,18 @@ DeBruijnGraph::node_index CanonicalDBG::reverse_complement(node_index node) cons
     if (node > offset_) {
         // we know that this node is definitely not present in the base graph
 
-        if (!primary_) {
-            rev_comp_cache_.Put(node - offset_, node);
-        } else if (!(graph_.get_k() & 1)) {
+        if (!(graph_.get_k() & 1))
```

Comment:
```suggestion
        if (graph_.get_k() % 2 == 0)
```

Reply:
Looks more natural to me

Reply:
And even better, compute it once and store in bool `k_odd_`. It won't make it any faster, but will make it easier to read.